internal/pkg/library/library.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2018-2019, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package library
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/sylabs/scs-library-client/client"
+)
+
+const defaultTag = "latest"
+
+type progressCallback func(int64, io.Reader, io.Writer) error
+
+// NormalizeLibraryRef strips off leading "library://" prefix, if any, and
+// appends the default tag (latest) if none specified.
+func NormalizeLibraryRef(libraryRef string) string {
+	var ir string
+	if strings.HasPrefix(libraryRef, "library://") {
+		ir = libraryRef[10:]
+	} else {
+		ir = libraryRef
+	}
+	if !strings.Contains(ir, ":") {
+		return ir + ":" + defaultTag
+	}
+	return ir
+}
+
+// DownloadImage is a helper function to wrap library image download operation
+func DownloadImage(ctx context.Context, c *client.Client, imagePath, libraryRef string, callback progressCallback) error {
+	// reassemble "stripped" library ref for scs-library-client
+	validLibraryRef := "library:///" + libraryRef
```

Comment:
I assume the three `/`'s is intentional here? Will we ever have a host in the reference? If so, this will make it part of the path correct?

Reply:
The `scs-library-client` supports URIs containing a host name, however it is not currently used. My intention here is to normalize the library ref on the Singularity side, such that the common Parse() function in the `scs-library-client` can be used going forward. I am not sure how/if this "enhanced" URI format will be used in Singularity at this point.